### PR TITLE
Improve "Open in Mastodon" by using Search API

### PR DESCRIPTION
**Pull request by kimar**
# Rationale

Previously the search action extension was failing on some occasion due to the JS generated DOM, of the links in question which the user might wanted to open. We're now leveraging the mastodon Search API to make sense of the web URLs.

### CODE REVIEW

File: OpenInActionExtension/ActionRequestHandler.swift [hunk 1]
```diff
@@ -56,13 +60,53 @@ class ActionRequestHandler: NSObject, NSExtensionRequestHandling {
...
+            let activeAuthenticationBox = Self.appContext
+                .authenticationService
+                .mastodonAuthenticationBoxes
+                .first
+        else {
+            return doneWithResults(nil)
+        }
+        
+        Mastodon.API
+            .V2
```
Comment (j-f1) [inline OpenInActionExtension/ActionRequestHandler.swift:76]: Is there a way to handle folks running old versions of Mastodon without the v2 search API?

Comment (zeitschlag) [reply OpenInActionExtension/ActionRequestHandler.swift:76]: If someone still runs [Mastodon with a version lower than 3.0](https://github.com/mastodon/mastodon/releases/tag/v3.0.0), they have way more severe issues than a broken app :D

File: OpenInActionExtension/ActionRequestHandler.swift [hunk 2]
```diff
@@ -9,13 +9,19 @@ import Combine
...
 import MastodonSDK
 import MastodonLocalization
 
 class ActionRequestHandler: NSObject, NSExtensionRequestHandling {
     var extensionContext: NSExtensionContext?
     var cancellables = [AnyCancellable]()
     
+    /// Capturing a static shared instance of AppContext here as otherwise there
+    /// will be lifecycle issues and we don't want to keep multiple AppContexts around
+    /// in case there another Action Extension process is spawned
```
Comment (zeitschlag) [inline OpenInActionExtension/ActionRequestHandler.swift:22]: 👍 for this comment!

**Review by zeitschlag — APPROVED** [review]
LGTM

**Pull request merged**